game-core/src/main/java/games/strategy/triplea/delegate/battle/BattleActions.java

```diff
@@ -59,4 +64,12 @@ Territory querySubmergeTerritory(
       GamePlayer retreatingPlayer,
       Collection<Territory> availableTerritories,
       String text);
+
+  default Player getRemotePlayer(final GamePlayer player, final IDelegateBridge bridge) {
+    // if its the null player, return a do nothing proxy
+    if (player.isNull()) {
+      return new WeakAi(player.getName());
```

Comment:
side-note, this dependency on `WeakAi` is something we really would like to get rid of:
- allows for all of weak ai to be deleted
- kinda poor design, the core game engine should not depend on AI code

Reply:
I grabbed this code from `AbstractBattle#getRemotePlayer`.  I'm not exactly sure why it is needed.

Reply:
I do recognize this code, this is one of the few and biggest dependencies on AI code from core-code IIRC. In essence we do this hack to supply a functioning player object and if my vague memory on this really holds for a default OOL algorithm. Using weakAi is a very ugly hack (which we've had for a long time; we can outright delete all of weak AI once we remove this dependency, which would be a significant simplification and removal of dead code).